classes/module/Module.php

```diff
@@ -1275,19 +1275,11 @@ public static function getModulesOnDisk($use_config = false, $logged_on_addons =
                         $file = substr($file, 0, -2);
                     }
 
-                    // We check any parse error before including the file.
-                    // If (false) is a trick to not load the class with "eval".
-                    // This way require_once will works correctly
-                    // But namespace and use statements need to be removed
-                    $content = preg_replace('/\n[\s\t]*?use\s.*?;/', '', $file);
-                    $content = preg_replace('/\n[\s\t]*?namespace\s.*?;/', '', $content);
                     try {
-                        if (substr(`php -l $file_path`, 0, 16) == 'No syntax errors' || eval('if (false){	'.$content.' }') !== false) {
-                            require_once(_PS_MODULE_DIR_.$module.'/'.$module.'.php');
-                        } else {
-                            throw new ParseError("Parse error");
-                        }
-                    } catch (ParseError $e) {
+                        $fileContents = _PS_MODULE_DIR_.$module.'/'.$module.'.php';
+                        $parser = (new PhpParser\ParserFactory)->create(PhpParser\ParserFactory::PREFER_PHP7);
+                        $parser->parse($fileContents);
+                    } catch (PhpParser\Error $e) {
```

Comment:
In the `try {}`, we need to require the module file, or it won't be found in the declared classes.

Please add 
```php
require_once($file_path);
```

or we will get several notices like `doge (missing class in /modules/doge/doge.php)`